usr/share/rear/layout/prepare/GNU/Linux/130_include_mount_filesystem_code.sh

```diff
@@ -69,10 +70,53 @@ mount_fs() {
+                # no special btrfs subvolumes setup is done which may lead to a falsely recreated system
+                # but we do what the user has explicity specified:
+                if is_false "$BTRFS_SUBVOLUME_GENERIC_SETUP" && is_false "$BTRFS_SUBVOLUME_SLES_SETUP" ; then
+                    LogPrint "Skipping btrfs subvolumes setup for $device on $mountpoint (BTRFS_SUBVOLUME_GENERIC_SETUP and BTRFS_SUBVOLUME_SLES_SETUP false)"
+                    continue
+                fi
+                if is_true "$BTRFS_SUBVOLUME_GENERIC_SETUP" ; then
+                    LogPrint "Doing generic btrfs subvolumes setup for $device on $mountpoint (BTRFS_SUBVOLUME_GENERIC_SETUP true)"
+                    btrfs_subvolumes_setup_generic $device $mountpoint $mountopts
+                    continue
+                fi
+                if is_true "$BTRFS_SUBVOLUME_SLES_SETUP" ; then
+                    LogPrint "Doing SLES-like btrfs subvolumes setup for $device on $mountpoint (BTRFS_SUBVOLUME_SLES_SETUP true)"
+                    btrfs_subvolumes_setup_SLES $device $mountpoint $mountopts
+                    continue
+                fi
+                if is_false "$BTRFS_SUBVOLUME_GENERIC_SETUP" ; then
+                    LogPrint "Doing SLES-like btrfs subvolumes setup for $device on $mountpoint (BTRFS_SUBVOLUME_GENERIC_SETUP false)"
+                    btrfs_subvolumes_setup_SLES $device $mountpoint $mountopts
+                    continue
+                fi
+                if is_false "$BTRFS_SUBVOLUME_SLES_SETUP" ; then
+                    LogPrint "Doing generic btrfs subvolumes setup for $device on $mountpoint (BTRFS_SUBVOLUME_SLES_SETUP false)"
+                    btrfs_subvolumes_setup_generic $device $mountpoint $mountopts
+                    continue
+                fi
+                if IsInArray "$device" ${BTRFS_SUBVOLUME_GENERIC_SETUP[@]} ; then
+                    LogPrint "Doing generic btrfs subvolumes setup for $device on $mountpoint (BTRFS_SUBVOLUME_GENERIC_SETUP contains $device)"
+                    btrfs_subvolumes_setup_generic $device $mountpoint $mountopts
+                    continue
+                fi
+                if IsInArray "$device" ${BTRFS_SUBVOLUME_SLES_SETUP[@]} ; then
+                    LogPrint "Doing SLES-like btrfs subvolumes setup for $device on $mountpoint (BTRFS_SUBVOLUME_SLES_SETUP contains $device)"
+                    btrfs_subvolumes_setup_SLES $device $mountpoint $mountopts
+                    continue
+                fi
+                # Final fallback to be backward compatible (btrfs_subvolumes_setup_SLES is the old way):
+                LogPrint "Doing SLES-like btrfs subvolumes setup for $device on $mountpoint (no match in BTRFS_SUBVOLUME_GENERIC_SETUP or BTRFS_SUBVOLUME_SLES_SETUP)"
+                btrfs_subvolumes_setup_SLES $device $mountpoint $mountopts
+            done
             ;;
```

Comment:
For me, it looks like this would work and support lots of possible combinations. (I'd probably just go for a simpler variant with less configurability.)

Reply:
I prefer to provide the user as much freedom to specify what ReaR should do
as possible to be implemented with reasonable effort.

E.g. one can specify
```
BTRFS_SUBVOLUME_SLES_SETUP=( 'false' '/dev/sda3' '/dev/sdc5' )
```
to avoid btrfs_subvolumes_setup_SLES for all devices
except '/dev/sda3' and '/dev/sdc5' which means
btrfs_subvolumes_setup_generic is done on all other devices.

Let's try out and see how things behave in practice for now.
If it gets too complicated I can simplify it as long as
it is not yet officially documented in default.conf.